### PR TITLE
Add optional `data` field to `SourceItem`

### PR DESCRIPTION
**Pull request by xuansontrinh**
relevant discussion: https://github.com/build-server-protocol/build-server-protocol/discussions/691

### CODE REVIEW

**Review by adpi2 — APPROVED** [review]
LGTM

File: spec/src/main/resources/META-INF/smithy/bsp/extensions/jvm.smithy [hunk 1]
```diff
@@ -150,4 +151,8 @@ list JvmCompileClasspathItems {
     member: JvmCompileClasspathItem
 }
 
-
+/// `JvmSourceItemData` contains jvm-specific metadata for a source item.
+@dataKind(kind: "jvm", extends: [SourceItemData])
+structure JvmSourceItemData {
+    packageName: String
```
Comment (agluszak) [inline spec/src/main/resources/META-INF/smithy/bsp/extensions/jvm.smithy:157]: Please document it further. Is it supposed to always match the package declaration in the source file?

**Review by agluszak — CHANGES_REQUESTED** [review]

**Comment by agluszak** [comment]
Also please add a changelog entry

File: spec/src/main/resources/META-INF/smithy/bsp/extensions/jvm.smithy [hunk 2]
```diff
@@ -150,4 +151,10 @@ list JvmCompileClasspathItems {
     member: JvmCompileClasspathItem
 }
 
-
+/// `JvmSourceItemData` contains jvm-specific metadata for a source item.
+@dataKind(kind: "jvm", extends: [SourceItemData])
+structure JvmSourceItemData {
+    /// The package name for the source item, must match the package declaration
+    /// if the source item is a file.
```
Comment (agluszak) [inline spec/src/main/resources/META-INF/smithy/bsp/extensions/jvm.smithy:158]: and if the source item is not a file? ;)

**Review by agluszak — APPROVED** [review]

**Pull request merged**